clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -301,6 +301,12 @@ public void send(Send send) {
             } catch (CancelledKeyException e) {
                 this.failedSends.add(connectionId);
                 close(channel, false);
+            } catch (Exception e) {
+                log.error("Unexpected exception during send, closing connection {} and rethrowing exception {}",
+                        connectionId, e);
+                channel.state(ChannelState.FAILED_SEND);
+                close(channel, false);
+                throw e;
```

Comment:
Do we think it makes sense to rethrow the exception here? Since this should only happen due to bugs, I think that is probably the way forward, but would be interested in other opinions.